### PR TITLE
Improvements for boleto (fees, fines and discount) and credit card payments (3DS 2.0)

### PR DESCRIPTION
**Pull request by romerosilva-meli**
this branch adds:

- Data models for processing boletos with fees, fines and discount
- Fields for processing credit card payments with 3DS 2.0

### CODE REVIEW

File: src/MercadoPago/Client/Payment/PaymentFeeRequest.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        /// <summary>
+        /// Type.
+        /// </summary>
+        public string Type { get; set; }
+
+        /// <summary>
+        /// Value.
+        /// </summary>
+        public decimal Value { get; set; }
+
```
Comment (rhames07) [inline src/MercadoPago/Client/Payment/PaymentFeeRequest.cs]:
```suggestion
```

File: src/MercadoPago/Client/Payment/PaymentRulesRequest.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Payment's rules.
+    /// </summary>
+    public class PaymentRulesRequest
+    {
+
```
Comment (rhames07) [inline src/MercadoPago/Client/Payment/PaymentRulesRequest.cs:11]:
```suggestion
```

File: src/MercadoPago/Resource/Payment/Payment.cs [hunk 3]
```diff
@@ -266,5 +266,21 @@ public class Payment : IResource
...
+
+        /// <summary>
+        /// Payment method.
+        /// </summary>
+        public PaymentMethod PaymentMethod { get; set; }
+
+        /// <summary>
+        /// 3DS Info.
+        ///</summary>
+        public PaymentThreeDSInfo ThreeDsinfo { get; set; }
```
Comment (rhames07) [inline src/MercadoPago/Resource/Payment/Payment.cs]: Não seria mais correto chamar de `ThreeDSInfo`?

**Review by rhames07 — APPROVED** [review]

File: src/MercadoPago.Tests/Client/Mock/CardPaymentWith3dsRequest.json [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        "fine":{
+          "type":"fixed",
+          "value":2
+        },
+        "interest":{
+          "type":"fixed",
+          "value":0.03
+        }
+      }
+    }
```
Comment (eltinMeli) [inline src/MercadoPago.Tests/Client/Mock/CardPaymentWith3dsRequest.json]: esses dados são referentes ao boleto, não precisa no 3ds

```suggestion
```

**Review by eltinMeli — APPROVED** [review]

**Pull request merged**